python/semantic_kernel/memory/storage/chroma_data_store.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from datetime import datetime
+from typing import TYPE_CHECKING, List, Optional
+
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.storage.data_entry import DataEntry
+from semantic_kernel.memory.storage.data_store_base import DataStoreBase
+
+if TYPE_CHECKING:
+    import chromadb
+    import chromadb.config
+    from chromadb.api.models.Collection import Collection
+    from chromadb.api.types import QueryResult
+
+
+def camel_to_snake(camel_str):
+    snake_str = camel_str[0].lower()
+    for char in camel_str[1:]:
+        if char.isupper():
+            snake_str += "_" + char.lower()
+        else:
+            snake_str += char
+    return snake_str
+
+
+class ChromaDataStore(DataStoreBase):
+    def __init__(
+        self,
+        persist_directory: Optional[str] = None,
+        client_settings: Optional["chromadb.config.Settings"] = None,
+    ) -> None:
+        try:
+            import chromadb
+            import chromadb.config
+            import chromadb.utils.embedding_functions as ef
+        except ImportError:
+            raise ValueError(
+                "Could not import chromadb python package. "
+                "Please install it with `pip install chromadb`."
+            )
+
+        if client_settings:
+            self._client_settings = client_settings
+        else:
+            self._client_settings = chromadb.config.Settings()
+            if persist_directory is not None:
+                self._client_settings = chromadb.config.Settings(
+                    chroma_db_impl="duckdb+parquet", persist_directory=persist_directory
+                )
+        self._client = chromadb.Client(self._client_settings)
+        self._persist_directory = persist_directory
+        self._default_embdding_function = ef.SentenceTransformerEmbeddingFunction()
+        self._default_query_includes = ["embeddings", "metadatas", "documents"]
+
+    async def get_collection_async(self, collection: str) -> Optional["Collection"]:
+        try:
+            # TODO: ChromeDB reject camel case collection names.
+            #   need to decide ChromaDataStore will do auto conversion inside or user takes responseibilities
+            collection_snake_case = camel_to_snake(collection)
```

Comment:
I think it's fine to keep the conversion internal to the memory store. This makes it interchangeable, though at small risk of container collision. I'd only really be concerned about this if the collection names were designed to be auto generated.

Reply:
If so, I'll leave this as a TODO and we can decide after chroma fixes the issue.